ratelimit/core.py

```diff
@@ -34,50 +36,69 @@ def __init__(
         backend: BaseBackend,
         config: Dict[str, Sequence[Rule]],
         *,
-        on_auth_error: Optional[Callable[[Exception], Awaitable[ASGIApp]]] = None,
-        on_blocked: Callable[[int], ASGIApp] = on_blocked,
+        on_auth_error: Optional[
+            Callable[[Exception], Awaitable[ASGIApp]]
+        ] = None,
+        on_blocked: Callable[[int], ASGIApp] = _on_blocked,
     ) -> None:
+
         self.app = app
         self.authenticate = authenticate
         self.backend = backend
-        self.config: Dict[re.Pattern, Sequence[Rule]] = {
-            re.compile(path): value for path, value in config.items()
-        }
+
+        assert isinstance(
+            self.authenticate, Callable
+        ), f"invalid authenticate function: {self.authenticate}"
+
+        assert isinstance(
+            backend, BaseBackend
+        ), f"invalid backend: {self.backend}"
+
+        try:
+            self.config: Dict[re.Pattern, Sequence[Rule]] = {
+                re.compile(path): value for path, value in config.items()
+            }
+        except re.error as exc:
```

Comment:
The capture here is redundant, I think `re.error` itself provides clear enough error information.